### PR TITLE
replace Pydantic parser with Schematic parser to make llama3:8b work

### PR DESCRIPTION
**Pull request by ernOho**
- provide simple output parser, which is compatible with opensource LLMs (where pydantic parser struggles)
- Adjust input prompts into evaluator

### CODE REVIEW

File: services/error_finders.py [hunk 1]
```diff
@@ -122,7 +122,16 @@ def create_prompt(self):
     def create_output_parser(self):
         """Creates pydantic model output parser with SummaryEvaluationItem"""
         # The parser that will look for the LLM output in my schema and return it back to me
-        self.output_parser = PydanticOutputParser(pydantic_object=SummaryEvaluationItem)
```
Comment (Bruno-val-bus) [inline services/error_finders.py:125]: How about we leave this line and the method as it is and implement your new response_schemas in a child class `SchemaSummaryChainWrapper(SummaryChainWrapper)`. 
Then we handle which class is instantiatiated `SchemaSummaryChainWrapper` or `SummaryChainWrapper)` in the factory based on which llm we are using.

This gives us control over the class we want to use at config level, not code level

File: services/error_finders.py [hunk 2]
```diff
@@ -122,7 +122,16 @@ def create_prompt(self):
...
+        # self.output_parser = PydanticOutputParser(pydantic_object=SummaryEvaluationItem)
+        response_schemas = [
```
Comment (Bruno-val-bus) [inline services/error_finders.py:126]: In `SchemaSummaryChainWrapper(SummaryChainWrapper)` we could write method that transforms the response to the pydantic class we intended to use originally.

**Comment by ernOho** [comment]
Ill have a look into it tomorrow

**Comment by Bruno-val-bus** [comment]
> Ill have a look into it tomorrow

I’ve been implementing a proposal. I’d suggest we discuss it first before you do any rework.

**Comment by ernOho** [comment]
Alright. I will have to wait for your proposal, test it with the local model before I can create another merge request relating to issue #6 .

**Comment by ernOho** [comment]
@Bruno-val-bus , I can't verify this pull request, as I created it.

Either you can review or accept it, or you need to create a new one I think.

**Comment by Bruno-val-bus** [comment]
> @Bruno-val-bus , I can't verify this pull request, as I created it.
> 
> Either you can review or accept it, or you need to create a new one I think.

Will review and accept then.
I noticed you used an abstract class for the configuration. I don’t see the configuration class behaving differently, so I’m not sure if an interface pattern is necessary here. Was there a special reason for it?

**Comment by ernOho** [comment]
Yeahr fair. No real special reason for it, exept for trying to follow the SOLID design principles (specifically Dependency Inversion principle). But yeah overkill or not implemented properly

**Comment by Bruno-val-bus** [comment]
Definitely a good idea to keep that design principle in mind. Whenever i decide wether to use an interface or not, I ask if the class is **behaving differently** or will only handle **different data** . For the configuration class i think it will merely do the latter. If you think the same, i will just keep the class and merge?

**Review by Bruno-val-bus — APPROVED** [review]
As discussed after merging `6-required-docker-files-to-launch-container-live-in-adequate-project-location` into `bug/ollama-working-with-langchain` and solving conflicts, it can be merged into master

**Pull request merged**